client/cmd/dexc-desktop/syncserver.go

```diff
@@ -0,0 +1,197 @@
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/client/app"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/encode"
+)
+
+const (
+	syncFilename = "syncfile"
+)
+
+// getSyncAddress: Get the address stored in the syncfile. If the file doesn't
+// exist, an empty string is returned with no error.
+func getSyncAddress(syncDir string) (addr, killKey string, err error) {
+	syncFile := filepath.Join(syncDir, syncFilename)
+	b, err := os.ReadFile(syncFile)
+	if err != nil && !os.IsNotExist(err) {
+		return "", "", err
+	}
+	if len(b) == 0 { // no sync file
+		return "", "", nil
+	}
+	lines := strings.Split(string(b), "\n")
+	if len(lines) != 2 {
+		return "", "", fmt.Errorf("unknown syncfile format: %q", string(b))
+	}
+	return lines[0], lines[1], nil
+}
+
+// sendKillSignal yeets a kill command to any address in the sync file.
+func sendKillSignal(syncDir string) {
+	addr, killKey, err := getSyncAddress(syncDir)
+	if err != nil {
+		log.Errorf("Error getting sync address for kill signal: %v", err)
+		return
+	}
+	if addr == "" {
+		log.Errorf("No sync address found for kill signal: %v", err)
+		return
+	}
+	resp, err := http.Post("http://"+addr+"/kill", "application/octet-stream", bytes.NewBufferString(killKey))
+	if err != nil {
+		log.Errorf("Error sending kill signal to sync server: %v", err)
+		return
+	}
+	if resp == nil {
+		log.Errorf("nil kill response")
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
+		log.Errorf("Unexpected response code from kill signal send: %d (%s)", resp.StatusCode, http.StatusText(resp.StatusCode))
+	}
+}
+
+// synchronize attempts to determine the state of any other dexc-desktop
+// instances that are running. If none are found, startServer will be true.
+// If another instance is found, close will be true, and this instance of
+// dexc-desktop should exit immediately.
+func synchronize(syncDir string) (startServer bool, err error) {
+	addr, _, err := getSyncAddress(syncDir)
+	if err != nil {
+		return false, err
+	}
+	if addr == "" {
+		// Just start the server then.
+		return true, nil
+	}
+	resp, err := http.Get("http://" + addr)
+	if err == nil {
+		if resp.StatusCode == http.StatusOK {
+			// Other instance will open the window.
+			return false, nil
+		} else {
+			return false, fmt.Errorf("sync server unexpected resonse code %d: %q", resp.StatusCode, http.StatusText(resp.StatusCode))
+		}
+	}
+	// Error will probably be something like "...: no route to host", but we'll
+	// just assume that any error means the server is not running.
+	return true, nil
+}
+
+// runServer runs an instance of the sync server. Received commands are
+// communicated via unbuffered channels. Blocking channels are ignored.
+func runServer(ctx context.Context, syncDir string, openC chan<- struct{}, killC chan<- os.Signal, netw dex.Network) {
+	host := app.DefaultHostByNetwork(netw)
+	l, err := net.Listen("tcp", host+":0")
```

Comment:
OK, don't get mad at me... I'm really sorry to flip like this, but the other loopback IPs are gonna be a hurdle for users where there OS does't make them aliases/usable out of the box.  For the syncserver, and since it has `:0`, I now think it's appropriate to just use "127.0.0.1", like you had it initially, only with the IPv4 address explicitly.  Sorry :grimacing: 
If we were dodging cookie issues with browsers, I'd say we make devs and testnet users deal with it, but there aren't any such cookie issues with the syncserver.